kafka-connect-coap/src/main/scala/com/datamountaineer/streamreactor/connect/coap/source/CoapReader.scala

```diff
@@ -0,0 +1,109 @@
+package com.datamountaineer.streamreactor.connect.coap.source
+
+import java.util.concurrent.LinkedBlockingQueue
+
+import akka.actor.{Actor, Props}
+import com.datamountaineer.streamreactor.connect.coap.DTLSConnectionFn
+import com.datamountaineer.streamreactor.connect.coap.configs.{CoapSourceSetting, CoapSourceSettings}
+import com.datamountaineer.streamreactor.connect.coap.domain.CoapMessageConverter
+import com.datamountaineer.streamreactor.connect.queues.QueueHelpers
+import com.typesafe.scalalogging.slf4j.StrictLogging
+import org.apache.kafka.common.config.ConfigException
+import org.apache.kafka.connect.source.SourceRecord
+import org.eclipse.californium.core.coap.Request
+import org.eclipse.californium.core.{CoapClient, CoapHandler, CoapResponse}
+import org.eclipse.californium.core.network.CoapEndpoint
+import org.eclipse.californium.core.network.config.NetworkConfig
+import org.eclipse.californium.scandium.DTLSConnector
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.collection.JavaConverters._
+
+/**
+  * Created by andrew@datamountaineer.com on 27/12/2016. 
+  * stream-reactor
+  */
+
+case object DataRequest
+case object StartChangeFeed
+case object StopChangeFeed
+case object Observing
+case object Discover
+
+object CoapReader {
+  def apply(settings: Set[CoapSourceSetting]): Map[String, Props] = {
+    settings.map(s => (s.kcql.getSource, Props(new CoapReader(s)))).toMap
+  }
+}
+
+case class CoapReader(setting: CoapSourceSetting) extends Actor with StrictLogging {
+  logger.info("Initialising COAP Reader")
+  val client = new CoapClient(setting.uri)
+  val buffer = new LinkedBlockingQueue[SourceRecord]
```

Comment:
this can be a simple arraylist. an actor is single threaded. this would add runtime overhead